src/main/java/org/altbeacon/beacon/BeaconManager.java

```diff
@@ -387,35 +387,93 @@ public boolean isBackgroundModeUninitialized() {
     }
 
     /**
-     * Specifies a class that should be called each time the <code>BeaconService</code> gets ranging
-     * data, which is nominally once per second when beacons are detected.
-     * <p/>
-     * IMPORTANT:  Only one RangeNotifier may be active for a given application.  If two different
-     * activities or services set different RangeNotifier instances, the last one set will receive
-     * all the notifications.
-     *
+     * allow to set Range Notifier
+     * Backward compatibility
      * @param notifier
-     * @see RangeNotifier
      */
-    public void setRangeNotifier(RangeNotifier notifier) {
-        rangeNotifier = notifier;
+    public void setRangeNotifier( RangeNotifier notifier )
+    {
+        addRangeNotifier( notifier );
```

Comment:
This could actually be backward compatible with the existing API if this method (and the monitoring equivalent) simply replaced the first notifier.